enclave-protocol/src/tdbe_protocol.rs

```diff
@@ -0,0 +1,35 @@
+use std::borrow::Cow;
+
+use parity_scale_codec::{Decode, Encode};
+
+use chain_core::tx::{data::TxId, TxWithOutputs};
+
+#[derive(Encode, Decode)]
+pub enum TdbeRequest<'a> {
+    /// Fetch all the transactions with outputs
+    GetTransactionsWithOutputs {
+        /// Transaction IDs for which to fetch all the transactions
+        transaction_ids: Cow<'a, [TxId]>,
+    },
+    /// Fetches keypackage for current node
+    GetKeyPackage,
+}
+
+#[derive(Encode, Decode)]
+pub enum TdbeResponse<'a> {
+    /// Contains all the requested transactions with outputs
+    GetTransactionsWithOutputs {
+        /// Requested transactions
+        transactions: Vec<TxWithOutputs>,
+    },
+    /// Contains keypackage for current node
+    GetKeyPackage {
+        /// Raw keypackage data
+        key_package: Vec<u8>, // TODO: Concrete `KeyPackage` type?
+    },
```

Comment:
this won't be in the necessary in the enclave-to-enclave response

Reply:
https://github.com/crypto-com/chain/pull/1940#discussion_r454789009